Automattic/pocket-casts-android
# Update Compose and Kotlin to the newest versions

## Description

This change updates:
* Kotlin to `1.9.22`.
* Compose to `1.6.1` using `2024.02.00` BOM
* Compose compiler to `1.5.9`
* AGP to `8.2.2`
* KSP to `1.9.22-1.0.17`
* And some other libraries like Horologist or Coil to follow the Compose updates

I also removed Accompanist library as APIs used by our app were deprecated and are now replaced by Compose equivalents. The biggest change is related to `com.google.accompanist.systemuicontroller.rememberSystemUiController()`. Instead we use `androidx.activity.enableEdgeToEdge()`. This cascaded in a lot of changes to the on-boarding UI.

## Testing Instructions

Main areas to test:
* Smoke test the app and image loading
* Test the on-boarding flow with different themes and check if status and navigation bars behave correctly when it comes to colors. [This](https://github.com/Automattic/pocket-casts-android/pull/656) might be a helpful PR.
* Smoke test the Wear OS app.

## Checklist
 
#### I have tested any UI changes...
